gnomad_qc/v4/sample_qc/assign_ancestry.py

```diff
@@ -367,60 +557,84 @@ def write_pca_results(
+        test = args.test
+        only_train_on_hgdp_tgp = args.only_train_on_hgdp_tgp
+
+        if (
+            args.max_number_mislabeled_training_samples is not None
+            and args.max_proportion_mislabeled_training_samples is not None
+        ):
+            raise ValueError(
+                "Only one of max_number_mislabeled_training_samples or"
+                " max_proportion_mislabeled_training_samples can be set"
+            )
+        elif args.max_proportion_mislabeled_training_samples is not None:
+            max_mislabeled = args.max_proportion_mislabeled_training_samples
+        elif args.max_number_mislabeled_training_samples is not None:
+            max_mislabeled = args.max_number_mislabeled_training_samples
+        else:
+            max_mislabeled = None
+
+        if args.run_pca:
+            pop_eigenvalues, pop_scores_ht, pop_loadings_ht = run_pca(
+                pca_related_samples_to_drop().ht(),
+                include_unreleasable_samples,
+                args.n_pcs,
+                test,
+            )
 
-    include_unreleasable_samples = args.include_unreleasable_samples
-    overwrite = args.overwrite
-    test = args.test
-    only_train_on_hgdp_tgp = args.only_train_on_hgdp_tgp
-
-    if args.run_pca:
-        pop_eigenvalues, pop_scores_ht, pop_loadings_ht = run_pca(
-            pca_related_samples_to_drop().ht(),
-            include_unreleasable_samples,
-            args.n_pcs,
-            test,
-        )
+            write_pca_results(
+                pop_eigenvalues,
+                pop_scores_ht,
+                pop_loadings_ht,
+                overwrite,
+                include_unreleasable_samples,
+                test,
+            )
 
-        write_pca_results(
-            pop_eigenvalues,
-            pop_scores_ht,
-            pop_loadings_ht,
-            overwrite,
-            include_unreleasable_samples,
-            test,
-        )
+        if args.assign_pops:
+            pop_pcs = args.pop_pcs
+            pop_pcs = list(range(1, pop_pcs[0] + 1)) if len(pop_pcs) == 1 else pop_pcs
+            logger.info("Using following PCs: %s", pop_pcs)
+            pop_ht, pops_rf_model = assign_pops(
+                args.min_pop_prob,
+                include_unreleasable_samples,
+                max_number_mislabeled_training_samples=args.max_number_mislabeled_training_samples,
+                max_proportion_mislabeled_training_samples=args.max_proportion_mislabeled_training_samples,
+                pcs=pop_pcs,
+                withhold_prop=args.withhold_prop,
+                test=test,
+                overwrite=overwrite,
+                only_train_on_hgdp_tgp=only_train_on_hgdp_tgp,
+                v4_population_spike=args.v4_population_spike,
+                v3_population_spike=args.v3_population_spike,
+            )
+            logger.info("Writing pop ht...")
+            pop_ht = pop_ht.checkpoint(
+                get_pop_ht(
```

Comment:
I think we need to modify `get_pop_ht` to return a finalized pop HT by default, so even if we had decided to train with only hgdp_tgp, that would only be included in the globals and by default when calling `get_pop_ht` you could still get the hgdp_tgp results if that is what we determined was the final pop training. It helps downstream scripts and future users of the resources if it's obvious what the final choice was